src/test/java/com/maxmind/db/MultiBufferTest.java

```diff
@@ -0,0 +1,338 @@
+        assertEquals(11, buffer.capacity());
+    }
+
+    @Test
+    public void testWrapInvalidChunkSize() {
+        ByteBuffer[] chunks = new ByteBuffer[] {
+                ByteBuffer.allocate(3),
+                ByteBuffer.allocate(8)
+        };
+
+        assertThrows(IllegalArgumentException.class, () -> new MultiBuffer(chunks, 8));
+    }
+
+    @Test
+    public void testReadFromFileChannel(@TempDir Path tempDir) throws IOException {
+        // Create test file
+        Path testFile = tempDir.resolve("test.dat");
+        byte[] testData = new byte[]{
+                (byte) 0xa2, 0x1, (byte) 0xf4,
+                (byte) 0xc2, 0x2a, 0x78,
+                0x2, 0x1, 0x1, (byte) 0xf4,
+                0x1, 0x7,
+                0x68, 0x40, 0x9, 0x21, (byte) 0xFB, 0x54, 0x44, 0x2E, (byte) 0xEA,
+        };
+        Files.write(testFile, testData);
+
+        try (FileChannel channel = FileChannel.open(testFile, StandardOpenOption.READ)) {
+            MultiBuffer buffer = new MultiBuffer(testData.length);
+            long bytesRead = buffer.readFrom(channel);
+            assertEquals(21, bytesRead);
+            assertEquals(21, buffer.position());
+        }
+    }
+
+    @Test
+    public void testMapFromChannel(@TempDir Path tempDir) throws IOException {
+        // Create test file
+        Path testFile = tempDir.resolve("test.dat");
+        byte[] testData = new byte[]{
+                (byte) 0xa2, 0x1, (byte) 0xf4,
+                (byte) 0xc2, 0x2a, 0x78,
+                0x2, 0x1, 0x1, (byte) 0xf4,
+                0x1, 0x7,
+                0x68, 0x40, 0x9, 0x21, (byte) 0xFB, 0x54, 0x44, 0x2E, (byte) 0xEA,
+        };
+        Files.write(testFile, testData);
+
+        try (FileChannel channel = FileChannel.open(testFile, StandardOpenOption.READ)) {
+            MultiBuffer buffer = MultiBuffer.mapFromChannel(channel);
+            assertEquals(21, buffer.capacity());
+        }
+    }
+
+    @Test
+    public void testMapFromEmptyChannel(@TempDir Path tempDir) throws IOException {
+        Path emptyFile = tempDir.resolve("empty.dat");
+        Files.createFile(emptyFile);
+
+        try (FileChannel channel = FileChannel.open(emptyFile, StandardOpenOption.READ)) {
+            assertThrows(IllegalArgumentException.class, () -> MultiBuffer.mapFromChannel(channel));
+        }
+    }
+
+    @Test
+    public void testDecodeString() throws CharacterCodingException {
+        MultiBuffer buffer = createBuffer(22);
+        buffer.position(26);
+        buffer.limit(29);
+        String result = buffer.decode(StandardCharsets.UTF_8.newDecoder());
+        assertEquals("123", result);
+        assertEquals(29, buffer.position());
+    }
+
+    @Test
+    public void testDecodeStringTooLarge() {
+        MultiBuffer buffer = new MultiBuffer((long) Integer.MAX_VALUE + 1, 1024);
+        assertThrows(IllegalStateException.class, () ->
+                buffer.decode(StandardCharsets.UTF_8.newDecoder()));
```

Comment:
I think this should be something like this so that we aren't creating a huge buffer:


```suggestion
       MultiBuffer buffer = createBuffer(65);
       buffer.position(62);
       buffer.limit(89);
       String result = buffer.decode(StandardCharsets.UTF_8.newDecoder());
       assertEquals("123456789012345678901234567", result);
```

Reply:
That would change the point of the test though, here I'm trying to test that `IllegaleStateException` is thrown when the `limit - position` is greated than max int. 🤔 

If I change it like you suggest that will never be thrown.

I added that check just because `CharBuffer` can't be bigger than max int like the other buffers. It would still fail but at least the failure is clearer like this.